pennylane/tape/tape.py

```diff
@@ -653,6 +653,23 @@ def inv(self):
 
         self._ops = list(reversed(self._ops))
 
+    def adjoint(self):
```

Comment:
We need to add an `adjoint()` method to the tape, so that the `qml.adjoint()` function knows what to do when it recursively hits a tape.